src/passes/I64ToI32Lowering.cpp

```diff
@@ -1526,7 +1520,35 @@ struct I64ToI32Lowering : public WalkerPass<PostWalker<I64ToI32Lowering>> {
   }
 
   void visitSelect(Select* curr) {
-    visitBranching<Select>(curr);
+    if (!hasOutParam(curr->ifTrue)) {
+      assert(!hasOutParam(curr->ifFalse));
+      return;
+    }
+    assert(hasOutParam(curr->ifFalse));
+    TempVar highBits = getTemp();
+    TempVar lowBits = getTemp();
+    TempVar cond = getTemp();
+    Block* result = builder->blockify(
+      builder->makeSetLocal(
+        lowBits,
+        builder->makeSelect(
+          builder->makeTeeLocal(cond, curr->condition),
```

Comment:
this tee violates the flat property of the IR. I think we should preserve it even if we don't depend on that right now. That means just doing a set earlier in the block being created here, and a get here.